qemu/tests/virtio_fs_supp_group_transfer.py

```diff
@@ -0,0 +1,99 @@
+import os.path
+
+from virttest import nfs, env_process
+from virttest import error_context, utils_disk, utils_misc
+
+from provider import virtio_fs_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Grant write access at users allowed by group permission
+    on shared dir( linux only )
+
+    1) Run the virtiofsd daemon on the host and Set log level to "debug".
+    2) Boot a guest on the host.
+    3) Log into guest with root, then mount the file system.
+    4) In the guest, add a new user(u1)
+       and set the user to the wheel group (as supplementary group)
+       and set the dir's group to 'wheel'
+    5) In guest, using root to create a test directory at the shared dir.
+    6) Run the basic io test with the u1 user
+    7) After test, clear the environment.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    username = params.get("new_guest_user")
+    fs_dest = params.get("fs_dest")
+    fs_target = params.get("fs_target")
+    fs_source_dir = params.get("fs_source_dir")
+    # nfs config
+    setup_local_nfs = params.get("setup_local_nfs")
+
+    testdir = "testdir"
+    guest_root_session = None
+    vm = None
+    nfs_local = None
+
+    try:
+        shared_dir = params.get("fs_source_dir")
+
+        if setup_local_nfs:
+            # delete the slash at the end
+            params["nfs_mount_dir"] = shared_dir[:-1]
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
+
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        vm = env.get_vm(params.get("main_vm"))
+        vm.verify_alive()
+        guest_root_session = vm.wait_for_login()
+
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, test.log.info)
+        if not utils_misc.make_dirs(fs_dest, session=guest_root_session):
+            test.fail("Creating directory was failed!")
+        error_context.context("Mount virtiofs target %s to %s inside"
+                              " guest." % (fs_target, fs_dest),
+                              test.log.info)
+        if not utils_disk.mount(fs_target, fs_dest, 'virtiofs',
+                                session=guest_root_session):
+            test.fail('Mount virtiofs target failed.')
+
+        guest_root_session.cmd(add_user_cmd % username)
```

Comment:
From the current debug log, there is no obviously critical steps

[stdlog] 2023-06-25 23:10:09,716 avocado.test INFO | Context: Create a destination directory /mnt/myfs inside guest.
[stdlog] 2023-06-25 23:10:09,716 aexpect.client DEBUG| [192.168.122.58] Sending command: mkdir -p /mnt/myfs
[stdlog] 2023-06-25 23:10:09,820 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:09,922 avocado.test INFO | Context: Mount virtiofs target myfs to /mnt/myfs inside guest.
[stdlog] 2023-06-25 23:10:09,922 aexpect.client DEBUG| [192.168.122.58] Sending command (safe): cat /proc/mounts
[stdlog] 2023-06-25 23:10:10,026 aexpect.client DEBUG| [192.168.122.58] Sending command (safe): mount -t virtiofs myfs /mnt/myfs
[stdlog] 2023-06-25 23:10:10,133 aexpect.client DEBUG| [192.168.122.58] Sending command (safe): echo $?
[stdlog] 2023-06-25 23:10:10,234 aexpect.client DEBUG| [192.168.122.58] Sending command: useradd user00001
[stdlog] 2023-06-25 23:10:11,345 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:11,446 aexpect.client DEBUG| [192.168.122.58] Sending command: usermod -G wheel user00001
[stdlog] 2023-06-25 23:10:11,722 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:11,824 aexpect.client DEBUG| [192.168.122.58] Sending command: cd /mnt/myfs && mkdir -m 770 testdir
[stdlog] 2023-06-25 23:10:11,931 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:12,033 aexpect.client DEBUG| [192.168.122.58] Sending command: cd /mnt/myfs && chgrp wheel testdir
[stdlog] 2023-06-25 23:10:12,197 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:12,299 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 240s)
[stdlog] 2023-06-25 23:10:12,300 avocado.virttest.virt_vm DEBUG| Found/Verified IP 192.168.122.58 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-06-25 23:10:13,780 aexpect.client DEBUG| [192.168.122.58] Sending command: su user00001
[stdlog] 2023-06-25 23:10:14,055 aexpect.client DEBUG| [192.168.122.58] Sending command: echo $?
[stdlog] 2023-06-25 23:10:14,157 avocado.test INFO | Context: Running viofs basic io test
[stdlog] 2023-06-25 23:10:14,157 avocado.test INFO | Context: The guest file in shared dir is /mnt/myfs/testdir/virtio_fs_test_file
[stdlog] 2023-06-25 23:10:14,157 avocado.test INFO | Context: Creating file under /mnt/myfs/testdir inside guest.

Could you add the critical steps description here?
error_context.context("Create a common user and change the group to wheelxxxxx")
And add another one in below code
error_context.context("Create a dir inside the virtiofs and change it's group to wheel")

error_context.context("Login the common user and try to write under the dir which belongs to wheel group")